Add `TelegramBotHandler` topics support

bernard-ng opened the pull request:
This PR adds support for the `topic` option in `TelegramBotHandler`, enabling the use of the Telegram message thread ID (`message_thread_id`). This unique identifier targets specific message threads (topics) within forum supergroups.

**Reference:**
- https://github.com/Seldaek/monolog/pull/1802
- [How to get the `message_thread_id`](https://stackoverflow.com/a/75178418)

bernard-ng commented:
@stof sorry for the ping, could you assist please ?

CI for PHP 7.4 fails with :
```
Required package "symfony/phpunit-bridge" is not present in the lock file.
This usually happens when composer files are incorrectly merged or the composer.json file is manually edited.
Read more about correctly resolving merge conflicts https://getcomposer.org/doc/articles/resolving-merge-conflicts.md
and prefer using the "require" command over editing the composer.json file directly https://getcomposer.org/doc/03-cli.md#require-r
Error: Process completed with exit code 4.
```

Is there any actions on my side to make theses workflows green ?

derrabus committed:
Drop support for Monolog 1 & 2

bernard-ng commented:
@stof @OskarStark any update on this ?

bernard-ng commented:
@stof @OskarStark any update on this ?

Amunak commented:
Hate to comment but I would also like to see this merged, Monoloh has had this for years now.

GromNaN committed:
Require PHP 8.2 and Symfony 7.3

GromNaN committed:
Merge branch '3.x' into 4.x

* 3.x:
  Reorganize files to match the "Reusable Bundles" structure

GromNaN committed:
Merge branch '3.x' into 4.x

* 3.x:
  Update changelog
  Use octal notation
  feat: console.interactive_only flag
  Add PR template
  Update Configuration.php

GromNaN committed:
feature #515 Require PHP 8.2, Symfony 7.3 and Monolog 3 (GromNaN)

This PR was merged into the 4.x branch.

Discussion
----------

Require PHP 8.2, Symfony 7.3 and Monolog 3

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | https://github.com/symfony/monolog-bundle/issues/511
| License       | MIT

Drop support for PHP < 8.2, Symfony < 7.3 and Monolog < 3

Commits
-------

50cc1da Require PHP 8.2 and Symfony 7.3

GromNaN commented:
Hi @bernard-ng, could you rebase on the branch 3.x? We updated the directory structure.

GromNaN committed:
feature #494 Drop support for Monolog < 3.5 (derrabus)

This PR was merged into the 4.x branch.

Discussion
----------

Drop support for Monolog < 3.5

Monolog 1 is more or less unmaintained these days. Dropping support for that old branch allows us to clean up the bundle a bit and reduce the size of our CI matrix. Projects that really want to keep using Monolog 1 can still install an old version of the bundle.

edit: I've also removed code paths exclusive to Monolog 2, given that version 3 is the minimum now.

Commits
-------

ca76996 Drop support for Monolog 1 & 2

GromNaN committed:
Merge branch '3.x' into 4.x

* 3.x:
  Run CI with various versions of Symfony
  Fix duplicate lines in composer.json
  Add 'exclude_fields' param to Slack handler

GromNaN committed:
Fix merge

HypeMC committed:
Remove legacy code paths

GromNaN committed:
minor #519 Remove legacy code paths (HypeMC)

This PR was merged into the 4.x branch.

Discussion
----------

Remove legacy code paths

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Removes some legacy code paths/conditions for older versions of Symfony, Monolog & PHP.

Commits
-------

4e72fad Remove legacy code paths

derrabus committed:
Run the tests with PHPUnit 11

GromNaN committed:
minor #520 Run the tests with PHPUnit 11 (derrabus)

This PR was merged into the 4.x branch.

Discussion
----------

Run the tests with PHPUnit 11

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

Let's run our test suite with the most recent PHPUnit version that we can use on PHP 8.2.

Commits
-------

2ca2fdf Run the tests with PHPUnit 11

GromNaN committed:
Merge branch '3.x' into 4.x

* 3.x:
  Don't add processor to all loggers if tags specify a channel or handler

HypeMC committed:
Minor cleanup

HypeMC committed:
Fix CI

GromNaN committed:
minor #521 Fix CI (HypeMC)

This PR was merged into the 4.x branch.

Discussion
----------

Fix CI

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Commits
-------

d324e73 Fix CI

GromNaN committed:
minor #522 Minor cleanup (HypeMC)

This PR was merged into the 4.x branch.

Discussion
----------

Minor cleanup

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Small follow-up to #519

Commits
-------

f9d6d7f Minor cleanup

derrabus committed:
Add native types everywhere

GromNaN committed:
feature #523 Add native types everywhere (derrabus)

This PR was merged into the 4.x branch.

Discussion
----------

Add native types everywhere

| Q             | A
| ------------- | ---
| Branch?       | 4.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

Let's use native types everywhere we can.

Commits
-------

7307356 Add native types everywhere

GromNaN committed:
Merge branch '3.x' into 4.x

* 3.x:
  Add Fabbot as a GitHub Actions
